# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/mips32/Lowering-mips32.cpp
# Commit: 69a27bde5003
# Full Hash: 69a27bde5003c70c97e8c31202c1fcf3e13d173e
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 09:30:56
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.4: Add tls dependency for WasmTruncateToInt64 and Int64ToFloatingPoint for arm. r=lth
#   
#   In this patch we add a tls dependency for the remaining nodes which use
#   BuiltinThunk to call c++ runtime. By ABI requirements WasmTlsReg should
#   be set.
# ==============================================================================

diff -r 9b42e0b690b6 -r 69a27bde5003 js/src/jit/mips32/Lowering-mips32.cpp
--- a/js/src/jit/mips32/Lowering-mips32.cpp	Wed Sep 09 09:50:35 2020 +0000
+++ b/js/src/jit/mips32/Lowering-mips32.cpp	Wed Sep 09 10:48:09 2020 +0000
@@ -234,6 +234,11 @@
                ins);
 }
 
+void LIRGeneratorMIPS::lowerWasmBuiltinTruncateToInt64(
+    MWasmBuiltinTruncateToInt64* ins) {
+  MOZ_CRASH("We don't use it for this architecture");
+}
+
 void LIRGenerator::visitInt64ToFloatingPoint(MInt64ToFloatingPoint* ins) {
   MDefinition* opd = ins->input();
   MOZ_ASSERT(opd->type() == MIRType::Int64);
@@ -242,3 +247,8 @@
   defineReturn(
       new (alloc()) LInt64ToFloatingPoint(useInt64RegisterAtStart(opd)), ins);
 }
+
+void LIRGeneratorMIPS::lowerBuiltinInt64ToFloatingPoint(
+    MBuiltinInt64ToFloatingPoint* ins) {
+  MOZ_CRASH("We don't use it for this architecture");
+}